Packs/CyberArk/Integrations/CyberArk/CyberArk.py

```diff
@@ -0,0 +1,934 @@
+        outputs_key_field='',
+        outputs=response
+    )
+    return_results(results)
+
+def update_safe_command(
+        client: Client,
+        safe_name: str,
+        safe_new_name: str = "",
+        description: str = "",
+        OLAC_enabled: str = "true",
+        managing_cmp: str = "",
+        number_of_versions_retention: str = "",
+        number_of_days_retention: str = "",
+        location: str = ""
+):
+    response = client.update_safe(safe_name, safe_new_name, description, OLAC_enabled, managing_cmp, number_of_versions_retention,
+                                  number_of_days_retention, location)
+    results = CommandResults(
+        outputs_prefix=f'CyberArk.{safe_name}',
+        outputs_key_field='',
+        outputs=response
+    )
+    return_results(results)
+
+
+def delete_safe_command(
+        client: Client,
+        safe_name: str,
+):
+    response = client.delete_safe(safe_name)
+    # the response should be an empty string, if an error raised it would be catch in the main block
+    # should never enter to the else block, extra precautions if something want wrong
+    if not response:
+        return_results(f"Safe {safe_name} was deleted")
+    else:
+        return_results(response)
+
+
+def list_safe_members_command(
+        client: Client,
+        safe_name: str
+):
+    response = client.list_safe_members(safe_name)
+    total_safe_members = response.get("Total")
+    demisto.log(f"There are {total_safe_members} safe members for {safe_name}")
+    results = CommandResults(
+        outputs_prefix=f'CyberArk.{safe_name}',
+        outputs_key_field='MemberName',
+        outputs=response.get("SafeMembers")
+    )
+    return_results(results)
+
+
+def add_safe_member_command(
+        client: Client,
+        safe_name: str,
+        member_name: str,
+        requests_authorization_level: str = "0",
+        membership_expiration_date: str = "",
+        permissions: str = "",
+        search_in: str = ""
+):
+    permissions_list = argToList(permissions)
+    response = client.add_safe_member(safe_name, member_name, requests_authorization_level, membership_expiration_date, permissions_list, search_in)
+    results = CommandResults(
+        outputs_prefix=f'CyberArk.{member_name}',
+        outputs_key_field='',
+        outputs=response.get("member")
+    )
+    return_results(results)
+
+
+def update_safe_member_command(
+        client: Client,
+        safe_name: str,
+        member_name: str,
+        requests_authorization_level: str = "0",
+        membership_expiration_date: str = "",
+        permissions: str = "",
+        search_in: str = ""
+):
+    permissions_list = argToList(permissions)
+    response = client.update_safe_member(safe_name, member_name, requests_authorization_level, membership_expiration_date, permissions_list)
+    results = CommandResults(
+        outputs_prefix=f'CyberArk.{member_name}',
+        outputs_key_field='',
```

Comment:
a safe does not have a unique id of some sort?

Reply:
no.. the safes commands are in the old api in which they don't get an Id (there are no relevant v2 commands)